ra_aid/agent_utils.py

```
@@ -56,58 +62,189 @@
             logger.debug("Using create_react_agent to instantiate agent.")
-            return create_react_agent(model, tools, checkpointer=checkpointer)
+            agent_kwargs = build_agent_kwargs(checkpointer, config, token_limit)
+            return create_react_agent(model, tools, **agent_kwargs)
         else:
-            logger.debug("Using CiaynAgent agent instance.")
-            return CiaynAgent(model, tools)
-            
+            logger.debug("Using CiaynAgent agent instance")
+            return CiaynAgent(model, tools, max_tokens=token_limit)
+
     except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)